Providers/Modules/nxOMSAuditdPlugin/Scripts/OMSAuditdPlugin.sh

```diff
@@ -0,0 +1,325 @@
+            cat $OMS_AUDIT_RULES_PATH
+            return $?
+        fi
+        return 0
+    else
+        cat $AUDIT_RULES_FILE | awk '\
+        BEGIN {
+            emit = 0;
+            start_found = 0;
+            end_found = 0;
+        }
+        {
+            if ($0 == "#### START OF OMS AUDIT RULES ####") { emit = 1; start_found = 1; next; }
+            if ($0 == "#### END OF OMS AUDIT RULES ####") { emit = 0; end_found = 1; next; }
+            if (emit == 1) { print; next; }
+        }
+        END {
+            if (start_found != end_found) { exit 1; }
+        }'
+        return $?
+    fi
+}
+
+remove_rules () {
+    if use_augenrules; then
+        if [ -e $OMS_AUDIT_RULES_PATH ]; then
+            rm -f $OMS_AUDIT_RULES_PATH
+            if [ $? -ne 0 ]; then
+                echo "Failed to remove $OMS_AUDIT_RULES_PATH" >&2
+                return 1
+            fi
+            /sbin/augenrules
+            if [ $? -ne 0 ]; then
+                echo "Failed to re-generate $AUDIT_RULES_FILE" >&2
+                return 1
+            fi
+        fi
+    else
+        TmpFile=$(mktemp /tmp/OMSAUditdPlugin.XXXXXXXX)
+        cat $AUDIT_RULES_FILE | awk '\
+        BEGIN {
+            emit = 1;
+            start_found = 0;
+            end_found = 0;
+        }
+        {
+            if ($0 == "#### START OF OMS AUDIT RULES ####") { emit = 0; start_found = 1; next; }
+            if ($0 == "#### END OF OMS AUDIT RULES ####") { emit = 1; end_found = 1; next; }
+            if (emit == 1) { print; next; }
+        }
+        END {
+            if (start_found != 1 || end_found != 1) { exit 1; }
+        }' > $TmpFile
+        if [ $? -ne 0 ]; then
+            echo "Failed to remove OMS audit rules from $AUDIT_RULES_FILE" >&2
+            rm -f $TmpFile
+            return 1
+        fi
+        cp $TmpFile $AUDIT_RULES_FILE
+        if [ $? -ne 0 ]; then
+            echo "Failed to edit $AUDIT_RULES_FILE" >&2
+            rm -f $TmpFile
+            return 1
+        fi
+        rm -f $TmpFile
+    fi
+}
+
+set_rules () {
+    if use_augenrules; then
+        cp $1 $OMS_AUDIT_RULES_PATH
+        if [ $? -ne 0 ]; then
+            echo "Failed to create $OMS_AUDIT_RULES_PATH" >&2
+            return 1
+        fi
+        chown root.root $OMS_AUDIT_RULES_PATH
+        if [ $? -ne 0 ]; then
+            echo "Failed to set ownership of $OMS_AUDIT_RULES_PATH" >&2
+            rm -f $OMS_AUDIT_RULES_PATH
+            return 1
+        fi
+        chmod 640 $OMS_AUDIT_RULES_PATH
+        if [ $? -ne 0 ]; then
+            echo "Failed to set permissions for $OMS_AUDIT_RULES_PATH" >&2
+            rm -f $OMS_AUDIT_RULES_PATH
+            return 1
+        fi
+        /sbin/augenrules
+        if [ $? -ne 0 ]; then
+            echo "Failed to re-generate $AUDIT_RULES_FILE" >&2
+            rm -f $OMS_AUDIT_RULES_PATH
+            return 1
+        fi
+    else
+        TmpOutFile=$(mktemp /tmp/OMSAUditdPlugin.XXXXXXXX)
```

Comment:
Is there an OMSAgent-specific or AuditD-specific tmp folder you could use instead of this? We can't necessarily assume access to /tmp/.

Also, you seem to have a type in OMS**AUditd**Plugin.XXXXXXXX